To merge develop branch to main

Surya-Santhosh opened the pull request:
The task enables the interactive reading and writing of data to AT24C02 EEPROM using ESP32, STM32 via I2C Communication. When write command with address and data entered in the console, it sent via UART to ESP32, then it sent to STM32 over I2C and the STM32 writes the value to the specified EEPROM address. When a read command is entered with an address, then it sent to STM32 over I2C and STM32 reads the data from AT24C02 and sends it back to the console for display. Two threads are created for master and slave operation and use semaphore for synchronisation.

Surya-Santhosh committed:
modify readme file add source file and header file

Surya-Santhosh committed:
modified readme file.

Surya-Santhosh committed:
Modified stm32.c - prefix of array name(ucRxData to pucRxData, ucTxData to pucTxData)

adarshtrenser2023 commented on Core/Inc/i2c/i2c.h:
```diff
@@ -0,0 +1,35 @@
+//************************* Global Variables **********************************
+
+//************************* Forward Declarations ******************************
+bool i2cWrite(I2C_HandleTypeDef *pstI2C,uint16 unDeviceAddress,
```
I think instead of keeping a long list of parameter you can generalise some variable into a structure

adarshtrenser2023 commented on Core/Inc/osLayer/osLayer.h:
```diff
@@ -0,0 +1,40 @@
+#include "common.h"
+
+//*************************** Global Types ************************************
+typedef struct THREAD
```
structure name should be like _THREAD_ please check coding standard document

adarshtrenser2023 commented on Core/Inc/stm32.h:
```diff
@@ -0,0 +1,50 @@
+bool stm32Slave();
+bool stm32Master();
+bool stm32SlaveI2C2Init();
+bool stm32SlaveI2C1Init();
+void StartTaskMaster();
+void StartTaskSlave();
+
```
i think these are i2c helpers for I2C so it should be in i2c.h

adarshtrenser2023 commented on Core/Src/i2c/i2c.c:
```diff
@@ -0,0 +1,88 @@
+// All Rights Reserved
+//*****************************************************************************
+// File    : i2c.c
+// Summary : Read and write operation for EEPROM.
```
I think this should be generic  i2c read and write is done for both EEPROM as well as ESp32 right?

adarshtrenser2023 commented on Core/Src/osLayer/osLayer.c:
```diff
@@ -0,0 +1,138 @@
+
+//***************************** Local Constants *******************************
+static THREAD stThread[] = {{0, {.priority = (osPriority_t) osPriorityNormal},
+		                    StartTaskSlave},
```
whar about  stack size?

adarshtrenser2023 commented on Core/Src/osLayer/osLayer.c:
```diff
@@ -0,0 +1,138 @@
+//******************************* Local Types *********************************
+
+//***************************** Local Constants *******************************
+static THREAD stThread[] = {{0, {.priority = (osPriority_t) osPriorityNormal},
```
i think thread id is a pointer , replace 0 by NULL

adarshtrenser2023 commented on Core/Src/osLayer/osLayer.c:
```diff
@@ -0,0 +1,138 @@
+{
+	uint8 ucIndex = 0;
+
+	for (ucIndex = 0; ucIndex < NUM_THREAD; ucIndex++)
```
instead of macro NUM_THREAD calculate using array size

adarshtrenser2023 commented on Core/Src/osLayer/osLayer.c:
```diff
@@ -0,0 +1,138 @@
+
+//***************************.osLayerGetSemHandler.***************************
+// Purpose : To provide semaphore handler for master and slave.
+// Inputs  : None
```
FHB mismatch with function parameter

adarshtrenser2023 commented on Core/Src/osLayer/osLayer.c:
```diff
@@ -0,0 +1,138 @@
+
+//*************************.osLayerMQueueCreation.*****************************
+// Purpose : To create message queue.
+// Inputs  : None
```
FHB not proper

adarshtrenser2023 commented on Core/Src/osLayer/osLayer.c:
```diff
@@ -0,0 +1,138 @@
+// Return  : true
+// Notes   : None
+//*****************************************************************************
+bool osLayerCreation()
```
Make it taskcreation

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,247 @@
+// All Rights Reserved
+//*****************************************************************************
+// File    : stm32.c
+// Summary : Read and write data from ESP32 to AT24C02.
```
use better description like helper for i2c communication with ESP32 and EEPROM

adarshtrenser2023 commented on Core/Src/osLayer/osLayer.c:
```diff
@@ -0,0 +1,138 @@
+		                    StartTaskMaster}};
+
+//***************************** Local Variables *******************************
+static osSemaphoreId_t semMasterHandle;
+static osSemaphoreId_t semSlaveHandle;
```
no prefix

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,247 @@
+I2C_HandleTypeDef hi2c1 = {0};
+
+//***************************** Local Constants *******************************
+static osSemaphoreId_t semMasterHandle;
+static osSemaphoreId_t semSlaveHandle;
+static osMessageQueueId_t mqSlaveHandle;
+static osMessageQueueId_t mqMasterHandle;
+
```
no prefix

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,247 @@
+#include "stm32.h"
+
+//******************************* Local Types *********************************
+I2C_HandleTypeDef hi2c2 = {0};
+I2C_HandleTypeDef hi2c1 = {0};
+
```
no prefix

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,247 @@
+// Return  : blResult
+// Notes   : None
+//*****************************************************************************
+bool stm32SlaveI2C1Init()
```
I think this is set as master?

adarshtrenser2023 reviewed (commented):
some APIs start with small letter some start with capital letter, 
Check with Aromal if we need to move stm32 code to i2c layer or not.

Check all functions FHB and also summary of file. It should be meaningful

Surya-Santhosh committed:
Modified code as per review comments.
add prefix for as per coding standard,
Modified FHB.

Surya-Santhosh committed:
stm32.c - Remove space

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,208 @@
+		printf("osLayerGetSemHandler failed\r\n");
+	}
+
+	if (true != i2cGetHandler(&stHi2c1, &stHi2c2))
```
why this handler need to be global?
Also common getHandler for both handler is not good create its own getHandler like I2c1GetHandler, I2c2GetHandler.
I think this funciton only need I2c2 handler.

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,208 @@
+
+	if (true != i2cGetHandler(&stHi2c1, &stHi2c2))
+	{
+		printf("osLayerGetI2CHandler failed\r\n");
+	}
+
+	if (HAL_OK == HAL_I2C_Slave_Receive(&stHi2c2, ucRxData, TX_MESSAGE_SIZE,
+			                            TIMEOUT))
+	{
+		memcpy(&stTxMessage, ucRxData, sizeof(ucRxData));
+
+		if (osOK == osMessageQueuePut((osMessageQueueId_t) pMqSlaveHandle,
+											   &stTxMessage, 0, TIMEOUT))
+		{
+			printf("Message Received from ESP32\r\n");
+		}
+	}
```
I think this function can be divided into two, this part as comm with ESp32 and the below part as comm with EEPROM
Also those functions can be directly called from task1. same with master.
Also this API FHB not mentioned about EEPROM

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,208 @@
+		printf("osLayerGetI2CHandler failed\r\n");
+	}
+
+	if (HAL_OK == HAL_I2C_Slave_Receive(&stHi2c2, ucRxData, TX_MESSAGE_SIZE,
```
why a direct HAL call? No wrapper?

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,208 @@
+		if (osOK == osMessageQueuePut((osMessageQueueId_t) pMqSlaveHandle,
+											   &stTxMessage, 0, TIMEOUT))
+		{
+			printf("Message Received from ESP32\r\n");
```
Is message sending or receiving print and function call are contradictory

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,208 @@
+	osSemaphoreRelease(pSemSlaveHandle);
+	osSemaphoreAcquire(pSemMasterHandle, osWaitForever);
+
+	if (osOK == osMessageQueueGet((osMessageQueueId_t) pMqMasterHandle,
```
dont we need a wrapper for message queue as well? check and update if needed

adarshtrenser2023 commented on Core/Src/stm32.c:
```diff
@@ -0,0 +1,208 @@
+
+	while (1)
+	{
+	  if (true != stm32Slave())
```
Already mentioned above this stm32Slave can be split into 2 function and can be called here directly

adarshtrenser2023 reviewed (commented):
Check all comments has been addressed seen some are not addressed.

Surya-Santhosh committed:
Modified the code as per the review comments.
i2c - Added wrapers for HAL function.
osLayer - Added wrapers for message queue put and get (osMessageQueuePut, osMessageQueueGet)
stm32 - split stm32slave() into 2 stm32slaveTransmitToEsp32(), stm32slaveReceivedFromEsp32()
